runners/spark/src/test/java/org/apache/beam/runners/spark/translation/streaming/SparkCoGroupByKeyStreamingTest.java

```diff
@@ -0,0 +1,152 @@
+                TimestampedValue.of(KV.of(2, 6), instant.plus(Duration.standardSeconds(1L))))
+            .advanceNextBatchWatermarkToInfinity();
+
+    CreateStream<KV<Integer, Integer>> source2 =
+        CreateStream.of(KvCoder.of(VarIntCoder.of(), VarIntCoder.of()), batchDuration())
+            .emptyBatch()
+            .advanceWatermarkForNextBatch(instant)
+            .nextBatch(
+                TimestampedValue.of(KV.of(1, 11), instant),
+                TimestampedValue.of(KV.of(1, 12), instant),
+                TimestampedValue.of(KV.of(1, 13), instant))
+            .advanceWatermarkForNextBatch(instant.plus(Duration.standardSeconds(1L)))
+            .nextBatch(
+                TimestampedValue.of(KV.of(2, 14), instant.plus(Duration.standardSeconds(1L))),
+                TimestampedValue.of(KV.of(2, 15), instant.plus(Duration.standardSeconds(1L))),
+                TimestampedValue.of(KV.of(2, 16), instant.plus(Duration.standardSeconds(1L))))
+            .advanceNextBatchWatermarkToInfinity();
+
+    PCollection<KV<Integer, Integer>> input1 =
+        pipeline
+            .apply("create source1", source1)
+            .apply(
+                "window input1",
+                Window.<KV<Integer, Integer>>into(FixedWindows.of(Duration.standardSeconds(3L)))
+                    .withAllowedLateness(Duration.ZERO));
+
+    PCollection<KV<Integer, Integer>> input2 =
+        pipeline
+            .apply("create source2", source2)
+            .apply(
+                "window input2",
+                Window.<KV<Integer, Integer>>into(FixedWindows.of(Duration.standardSeconds(3L)))
+                    .withAllowedLateness(Duration.ZERO));
+
+    PCollection<KV<Integer, CoGbkResult>> output =
+        KeyedPCollectionTuple.of(INPUT1_TAG, input1)
+            .and(INPUT2_TAG, input2)
+            .apply(CoGroupByKey.create());
+    PAssert.that("Wrong output of the join using CoGroupByKey in streaming mode", output)
+        .satisfies(
+            (SerializableFunction<Iterable<KV<Integer, CoGbkResult>>, Void>)
+                input -> {
+                  assertEquals("Wrong size of the output PCollection", 2, Iterables.size(input));
+                  for (KV<Integer, CoGbkResult> element : input) {
+                    if (element.getKey() == 1) {
+                      assertThat(
+                          "Elements of PCollection input1 for key \"1\" are not present in the output PCollection",
+                          element.getValue().getAll(INPUT1_TAG),
+                          containsInAnyOrder(1, 2, 3));
+                      assertThat(
+                          "Elements of PCollection input2 for key \"1\" are not present in the output PCollection",
+                          element.getValue().getAll(INPUT2_TAG),
+                          containsInAnyOrder(11, 12, 13));
+                    } else if (element.getKey() == 2) {
+                      assertThat(
```

Comment:
The same comment about value collection size check.

Reply:
+1